### PR TITLE
Add picture service, endpoints and Azure Blob integration

### PR DESCRIPTION
**Pull request by Patonymous**

### CODE REVIEW

**Comment by Patonymous** [comment]
I will reopen this PR when EF migrations issues are fixed

**Comment by Patonymous** [comment]
EF migrations were fixed, but as I'm no expert, I won't be surprised if in the future we get errors again. I suggest to reset migrations soon

File: Api/Controllers/PicturesController.cs [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+            {
+                return BadRequest(e.Message);
+            }
+        }
+
+        [HttpPost]
+        [Route("event/{target}")]
+        [Produces("text/plain")]
+        [Authorize]
+        public ActionResult<string> PostEventPicture([FromRoute] Guid target, [FromBody] IFormFile picture, [FromQuery] int index)
```
Comment (Luuxo77) [inline Api/Controllers/PicturesController.cs]: I don't know if it is a problem on my side, but uploading picture through Swagger doesn't work for me, as it says that "The picture field is required". After removing [FromBody] it works.

File: Domain/Services/Implementations/PictureService.cs
Comment (Luuxo77) [inline Domain/Services/Implementations/PictureService.cs:1]: I think there is a problem with checking max pictures limits, as by specifying index as 0 in every request while uploading post pictures I was able to upload more than 3 pictures.

File: Api/Controllers/PicturesController.cs (hunk 1 above)
Comment (Patonymous) [reply Api/Controllers/PicturesController.cs]: I know about this, wasted an hour trying to get it work on /user/profile endpoint, that's why there I removed [FromBody]. I simply forgot to remove it in the other two. Will fix

File: Domain/Services/Implementations/PictureService.cs
Comment (Patonymous) [reply Domain/Services/Implementations/PictureService.cs:1]: You're right, I switched to different idea mid-coding and didn't correct the logic. Originally I thought that uploading at index "i" should overwrite image "i". Eventually I decided that it should move all pictures (i->i+1) and insert at "i". Will fix

File: Api/MinispaceExceptionHandler.cs
Comment (Luuxo77) [inline Api/MinispaceExceptionHandler.cs:1]: StorageException (and any storage related exception) won't get caught in global handling as they do not inherit from MiniSpaceException (first line of exception handler makes sure that the exception is MiniSpaceException).

File: Domain/Services/Exceptions.cs
Comment (Luuxo77) [inline Domain/Services/Exceptions.cs:1]: As mentioned in exception handler.

**Review by Luuxo77 — COMMENTED** [review]
Everything except a small thing about exceptions looks good to me.

File: Api/MinispaceExceptionHandler.cs
Comment (Patonymous) [reply Api/MinispaceExceptionHandler.cs:1]: Thx, good catch, done

**Review by Luuxo77 — APPROVED** [review]

**Pull request merged**